File: mod_periodic_table.c
#include <stdio.h>

#define C_RESET "\x1b[0m"
#define C_BLUE "\x1b[34m"
#define C_GREEN "\x1b[32m"
#define C_RED "\x1b[31m"

typedef struct
{
    int atomic_number;
    const char *symbol;
    const char *name;
} Element;

static const Element elements[] = {
    {1, "H", "Hydrogen"}, {2, "He", "Helium"}, {3, "Li", "Lithium"}, {4, "Be", "Beryllium"},
    {5, "B", "Boron"}, {6, "C", "Carbon"}, {7, "N", "Nitrogen"}, {8, "O", "Oxygen"},
    {9, "F", "Fluorine"}, {10, "Ne", "Neon"}, {11, "Na", "Sodium"}, {12, "Mg", "Magnesium"},
    {13, "Al", "Aluminium"}, {14, "Si", "Silicon"}, {15, "P", "Phosphorus"}, {16, "S", "Sulfur"},
    {17, "Cl", "Chlorine"}, {18, "Ar", "Argon"}, {19, "K", "Potassium"}, {20, "Ca", "Calcium"},
    {21, "Sc", "Scandium"}, {22, "Ti", "Titanium"}, {23, "V", "Vanadium"}, {24, "Cr", "Chromium"},
    {25, "Mn", "Manganese"}, {26, "Fe", "Iron"}, {27, "Co", "Cobalt"}, {28, "Ni", "Nickel"},
    {29, "Cu", "Copper"}, {30, "Zn", "Zinc"}, {31, "Ga", "Gallium"}, {32, "Ge", "Germanium"},
    {33, "As", "Arsenic"}, {34, "Se", "Selenium"}, {35, "Br", "Bromine"}, {36, "Kr", "Krypton"},
    {37, "Rb", "Rubidium"}, {38, "Sr", "Strontium"}, {39, "Y", "Yttrium"}, {40, "Zr", "Zirconium"},
    {41, "Nb", "Niobium"}, {42, "Mo", "Molybdenum"}, {43, "Tc", "Technetium"}, {44, "Ru", "Ruthenium"},
    {45, "Rh", "Rhodium"}, {46, "Pd", "Palladium"}, {47, "Ag", "Silver"}, {48, "Cd", "Cadmium"},
    {49, "In", "Indium"}, {50, "Sn", "Tin"}, {51, "Sb", "Antimony"}, {52, "Te", "Tellurium"},
    {53, "I", "Iodine"}, {54, "Xe", "Xenon"}, {55, "Cs", "Cesium"}, {56, "Ba", "Barium"},
    {57, "La", "Lanthanum"}, {58, "Ce", "Cerium"}, {59, "Pr", "Praseodymium"}, {60, "Nd", "Neodymium"},
    {61, "Pm", "Promethium"}, {62, "Sm", "Samarium"}, {63, "Eu", "Europium"}, {64, "Gd", "Gadolinium"},
    {65, "Tb", "Terbium"}, {66, "Dy", "Dysprosium"}, {67, "Ho", "Holmium"}, {68, "Er", "Erbium"},
    {69, "Tm", "Thulium"}, {70, "Yb", "Ytterbium"}, {71, "Lu", "Lutetium"}, {72, "Hf", "Hafnium"},
    {73, "Ta", "Tantalum"}, {74, "W", "Tungsten"}, {75, "Re", "Rhenium"}, {76, "Os", "Osmium"},
    {77, "Ir", "Iridium"}, {78, "Pt", "Platinum"}, {79, "Au", "Gold"}, {80, "Hg", "Mercury"},
    {81, "Tl", "Thallium"}, {82, "Pb", "Lead"}, {83, "Bi", "Bismuth"}, {84, "Po", "Polonium"},
    {85, "At", "Astatine"}, {86, "Rn", "Radon"}, {87, "Fr", "Francium"}, {88, "Ra", "Radium"},
    {89, "Ac", "Actinium"}, {90, "Th", "Thorium"}, {91, "Pa", "Protactinium"}, {92, "U", "Uranium"},
    {93, "Np", "Neptunium"}, {94, "Pu", "Plutonium"}, {95, "Am", "Americium"}, {96, "Cm", "Curium"},
    {97, "Bk", "Berkelium"}, {98, "Cf", "Californium"}, {99, "Es", "Einsteinium"}, {100, "Fm", "Fermium"},
    {101, "Md", "Mendelevium"}, {102, "No", "Nobelium"}, {103, "Lr", "Lawrencium"}, {104, "Rf", "Rutherfordium"},
    {105, "Db", "Dubnium"}, {106, "Sg", "Seaborgium"}, {107, "Bh", "Bohrium"}, {108, "Hs", "Hassium"},
    {109, "Mt", "Meitnerium"}, {110, "Ds", "Darmstadtium"}, {111, "Rg", "Roentgenium"}, {112, "Cn", "Copernicium"},
    {113, "Nh", "Nihonium"}, {114, "Fl", "Flerovium"}, {115, "Mc", "Moscovium"}, {116, "Lv", "Livermorium"},
    {117, "Ts", "Tennessine"}, {118, "Og", "Oganesson"}};

static const int element_count = (int)(sizeof(elements) / sizeof(elements[0]));

static const Element *find_element_by_atomic_number(int atomic_number)
{
    int i;
    for (i = 0; i < element_count; i++)
    {
        if (elements[i].atomic_number == atomic_number)
        {
            return &elements[i];
        }
    }
    return NULL;
}

static void print_element_card(const Element *e)
{
    printf(C_GREEN "\n+---------------- ELEMENT CARD ----------------+\n" C_RESET);
    printf("| Atomic Number : %-28d|\n", e->atomic_number);
    printf("| Symbol        : %-28s|\n", e->symbol);
    printf("| Name          : %-28s|\n", e->name);
    printf(C_GREEN "+----------------------------------------------+\n\n" C_RESET);
}

int main(void)
{
    int choice;

    printf(C_BLUE "========================================================\n" C_RESET);
    printf(C_BLUE "                MODERN PERIODIC TABLE HUB\n" C_RESET);
    printf(C_BLUE "========================================================\n" C_RESET);
    printf("Total elements indexed: %d\n\n", element_count);

    while (1)
    {
        int atomic_number;
        const Element *element;

        printf("1. Search by atomic number\n");
        printf("2. Exit\n");
        printf("Select: ");

        if (scanf("%d", &choice) != 1)
        {
            printf(C_RED "Invalid input. Closing app.\n" C_RESET);
            return 0;
        }

        if (choice == 2)
        {
            printf("Thanks for exploring chemistry.\n");
            break;
        }

        if (choice != 1)
        {
            printf(C_RED "Choose 1 or 2 only.\n\n" C_RESET);
            continue;
        }

        printf("Enter atomic number (1-118): ");
        if (scanf("%d", &atomic_number) != 1)
        {
            printf(C_RED "Invalid value. Closing app.\n" C_RESET);
            return 0;
        }

        element = find_element_by_atomic_number(atomic_number);
        if (element == NULL)
        {
            printf(C_RED "No element found for %d\n\n" C_RESET, atomic_number);
            continue;
        }

        print_element_card(element);
    }

    return 0;
}

File: mod_periodic_table.py
C_RESET = "\x1b[0m"
C_BLUE = "\x1b[34m"
C_GREEN = "\x1b[32m"
C_RED = "\x1b[31m"


ELEMENTS = [
    {"atomic_number": 1, "symbol": "H", "name": "Hydrogen"},
    {"atomic_number": 2, "symbol": "He", "name": "Helium"},
    {"atomic_number": 3, "symbol": "Li", "name": "Lithium"},
    {"atomic_number": 4, "symbol": "Be", "name": "Beryllium"},
    {"atomic_number": 5, "symbol": "B", "name": "Boron"},
    {"atomic_number": 6, "symbol": "C", "name": "Carbon"},
    {"atomic_number": 7, "symbol": "N", "name": "Nitrogen"},
    {"atomic_number": 8, "symbol": "O", "name": "Oxygen"},
    {"atomic_number": 9, "symbol": "F", "name": "Fluorine"},
    {"atomic_number": 10, "symbol": "Ne", "name": "Neon"},
    {"atomic_number": 11, "symbol": "Na", "name": "Sodium"},
    {"atomic_number": 12, "symbol": "Mg", "name": "Magnesium"},
    {"atomic_number": 13, "symbol": "Al", "name": "Aluminium"},
    {"atomic_number": 14, "symbol": "Si", "name": "Silicon"},
    {"atomic_number": 15, "symbol": "P", "name": "Phosphorus"},
    {"atomic_number": 16, "symbol": "S", "name": "Sulfur"},
    {"atomic_number": 17, "symbol": "Cl", "name": "Chlorine"},
    {"atomic_number": 18, "symbol": "Ar", "name": "Argon"},
    {"atomic_number": 19, "symbol": "K", "name": "Potassium"},
    {"atomic_number": 20, "symbol": "Ca", "name": "Calcium"},
    {"atomic_number": 21, "symbol": "Sc", "name": "Scandium"},
    {"atomic_number": 22, "symbol": "Ti", "name": "Titanium"},
    {"atomic_number": 23, "symbol": "V", "name": "Vanadium"},
    {"atomic_number": 24, "symbol": "Cr", "name": "Chromium"},
    {"atomic_number": 25, "symbol": "Mn", "name": "Manganese"},
    {"atomic_number": 26, "symbol": "Fe", "name": "Iron"},
    {"atomic_number": 27, "symbol": "Co", "name": "Cobalt"},
    {"atomic_number": 28, "symbol": "Ni", "name": "Nickel"},
    {"atomic_number": 29, "symbol": "Cu", "name": "Copper"},
    {"atomic_number": 30, "symbol": "Zn", "name": "Zinc"},
    {"atomic_number": 31, "symbol": "Ga", "name": "Gallium"},
    {"atomic_number": 32, "symbol": "Ge", "name": "Germanium"},
    {"atomic_number": 33, "symbol": "As", "name": "Arsenic"},
    {"atomic_number": 34, "symbol": "Se", "name": "Selenium"},
    {"atomic_number": 35, "symbol": "Br", "name": "Bromine"},
    {"atomic_number": 36, "symbol": "Kr", "name": "Krypton"},
    {"atomic_number": 37, "symbol": "Rb", "name": "Rubidium"},
    {"atomic_number": 38, "symbol": "Sr", "name": "Strontium"},
    {"atomic_number": 39, "symbol": "Y", "name": "Yttrium"},
    {"atomic_number": 40, "symbol": "Zr", "name": "Zirconium"},
    {"atomic_number": 41, "symbol": "Nb", "name": "Niobium"},
    {"atomic_number": 42, "symbol": "Mo", "name": "Molybdenum"},
    {"atomic_number": 43, "symbol": "Tc", "name": "Technetium"},
    {"atomic_number": 44, "symbol": "Ru", "name": "Ruthenium"},
    {"atomic_number": 45, "symbol": "Rh", "name": "Rhodium"},
    {"atomic_number": 46, "symbol": "Pd", "name": "Palladium"},
    {"atomic_number": 47, "symbol": "Ag", "name": "Silver"},
    {"atomic_number": 48, "symbol": "Cd", "name": "Cadmium"},
    {"atomic_number": 49, "symbol": "In", "name": "Indium"},
    {"atomic_number": 50, "symbol": "Sn", "name": "Tin"},
    {"atomic_number": 51, "symbol": "Sb", "name": "Antimony"},
    {"atomic_number": 52, "symbol": "Te", "name": "Tellurium"},
    {"atomic_number": 53, "symbol": "I", "name": "Iodine"},
    {"atomic_number": 54, "symbol": "Xe", "name": "Xenon"},
    {"atomic_number": 55, "symbol": "Cs", "name": "Cesium"},
    {"atomic_number": 56, "symbol": "Ba", "name": "Barium"},
    {"atomic_number": 57, "symbol": "La", "name": "Lanthanum"},
    {"atomic_number": 58, "symbol": "Ce", "name": "Cerium"},
    {"atomic_number": 59, "symbol": "Pr", "name": "Praseodymium"},
    {"atomic_number": 60, "symbol": "Nd", "name": "Neodymium"},
    {"atomic_number": 61, "symbol": "Pm", "name": "Promethium"},
    {"atomic_number": 62, "symbol": "Sm", "name": "Samarium"},
    {"atomic_number": 63, "symbol": "Eu", "name": "Europium"},
    {"atomic_number": 64, "symbol": "Gd", "name": "Gadolinium"},
    {"atomic_number": 65, "symbol": "Tb", "name": "Terbium"},
    {"atomic_number": 66, "symbol": "Dy", "name": "Dysprosium"},
    {"atomic_number": 67, "symbol": "Ho", "name": "Holmium"},
    {"atomic_number": 68, "symbol": "Er", "name": "Erbium"},
    {"atomic_number": 69, "symbol": "Tm", "name": "Thulium"},
    {"atomic_number": 70, "symbol": "Yb", "name": "Ytterbium"},
    {"atomic_number": 71, "symbol": "Lu", "name": "Lutetium"},
    {"atomic_number": 72, "symbol": "Hf", "name": "Hafnium"},
    {"atomic_number": 73, "symbol": "Ta", "name": "Tantalum"},
    {"atomic_number": 74, "symbol": "W", "name": "Tungsten"},
    {"atomic_number": 75, "symbol": "Re", "name": "Rhenium"},
    {"atomic_number": 76, "symbol": "Os", "name": "Osmium"},
    {"atomic_number": 77, "symbol": "Ir", "name": "Iridium"},
    {"atomic_number": 78, "symbol": "Pt", "name": "Platinum"},
    {"atomic_number": 79, "symbol": "Au", "name": "Gold"},
    {"atomic_number": 80, "symbol": "Hg", "name": "Mercury"},
    {"atomic_number": 81, "symbol": "Tl", "name": "Thallium"},
    {"atomic_number": 82, "symbol": "Pb", "name": "Lead"},
    {"atomic_number": 83, "symbol": "Bi", "name": "Bismuth"},
    {"atomic_number": 84, "symbol": "Po", "name": "Polonium"},
    {"atomic_number": 85, "symbol": "At", "name": "Astatine"},
    {"atomic_number": 86, "symbol": "Rn", "name": "Radon"},
    {"atomic_number": 87, "symbol": "Fr", "name": "Francium"},
    {"atomic_number": 88, "symbol": "Ra", "name": "Radium"},
    {"atomic_number": 89, "symbol": "Ac", "name": "Actinium"},
    {"atomic_number": 90, "symbol": "Th", "name": "Thorium"},
    {"atomic_number": 91, "symbol": "Pa", "name": "Protactinium"},
    {"atomic_number": 92, "symbol": "U", "name": "Uranium"},
    {"atomic_number": 93, "symbol": "Np", "name": "Neptunium"},
    {"atomic_number": 94, "symbol": "Pu", "name": "Plutonium"},
    {"atomic_number": 95, "symbol": "Am", "name": "Americium"},
    {"atomic_number": 96, "symbol": "Cm", "name": "Curium"},
    {"atomic_number": 97, "symbol": "Bk", "name": "Berkelium"},
    {"atomic_number": 98, "symbol": "Cf", "name": "Californium"},
    {"atomic_number": 99, "symbol": "Es", "name": "Einsteinium"},
    {"atomic_number": 100, "symbol": "Fm", "name": "Fermium"},
    {"atomic_number": 101, "symbol": "Md", "name": "Mendelevium"},
    {"atomic_number": 102, "symbol": "No", "name": "Nobelium"},
    {"atomic_number": 103, "symbol": "Lr", "name": "Lawrencium"},
    {"atomic_number": 104, "symbol": "Rf", "name": "Rutherfordium"},
    {"atomic_number": 105, "symbol": "Db", "name": "Dubnium"},
    {"atomic_number": 106, "symbol": "Sg", "name": "Seaborgium"},
    {"atomic_number": 107, "symbol": "Bh", "name": "Bohrium"},
    {"atomic_number": 108, "symbol": "Hs", "name": "Hassium"},
    {"atomic_number": 109, "symbol": "Mt", "name": "Meitnerium"},
    {"atomic_number": 110, "symbol": "Ds", "name": "Darmstadtium"},
    {"atomic_number": 111, "symbol": "Rg", "name": "Roentgenium"},
    {"atomic_number": 112, "symbol": "Cn", "name": "Copernicium"},
    {"atomic_number": 113, "symbol": "Nh", "name": "Nihonium"},
    {"atomic_number": 114, "symbol": "Fl", "name": "Flerovium"},
    {"atomic_number": 115, "symbol": "Mc", "name": "Moscovium"},
    {"atomic_number": 116, "symbol": "Lv", "name": "Livermorium"},
    {"atomic_number": 117, "symbol": "Ts", "name": "Tennessine"},
    {"atomic_number": 118, "symbol": "Og", "name": "Oganesson"},
]


def find_element_by_atomic_number(atomic_number):
    for element in ELEMENTS:
        if element["atomic_number"] == atomic_number:
            return element
    return None


def print_element_card(element):
    print(f"{C_GREEN}\n+---------------- ELEMENT CARD ----------------+{C_RESET}")
    print(f"| Atomic Number : {element['atomic_number']:<28}|")
    print(f"| Symbol        : {element['symbol']:<28}|")
    print(f"| Name          : {element['name']:<28}|")
    print(f"{C_GREEN}+----------------------------------------------+\n{C_RESET}")


def main():
    element_count = len(ELEMENTS)

    print(f"{C_BLUE}========================================================{C_RESET}")
    print(f"{C_BLUE}                MODERN PERIODIC TABLE HUB{C_RESET}")
    print(f"{C_BLUE}========================================================{C_RESET}")
    print(f"Total elements indexed: {element_count}\n")

    while True:
        print("1. Search by atomic number")
        print("2. Exit")

        try:
            choice = int(input("Select: "))
        except ValueError:
            print(f"{C_RED}Invalid input. Closing app.{C_RESET}")
            return

        if choice == 2:
            print("Thanks for exploring chemistry.")
            break

        if choice != 1:
            print(f"{C_RED}Choose 1 or 2 only.\n{C_RESET}")
            continue

        try:
            atomic_number = int(input("Enter atomic number (1-118): "))
        except ValueError:
            print(f"{C_RED}Invalid value. Closing app.{C_RESET}")
            return

        element = find_element_by_atomic_number(atomic_number)
        if element is None:
            print(f"{C_RED}No element found for {atomic_number}\n{C_RESET}")
            continue

        print_element_card(element)


if __name__ == "__main__":
    main()
